src/protect-api/Program.cs

```diff
@@ -1,26 +1,37 @@
+// <ms_docref_import_types>
 using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Authentication.JwtBearer;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.Identity.Web;
+// </ms_docref_import_types>
 
 var builder = WebApplication.CreateBuilder(args);
 
+// <ms_docref_add_msal>
 builder.Services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
                 .AddMicrosoftIdentityWebApi(builder.Configuration.GetSection("AzureAd"));
+// </ms_docref_add_msal>
+// <ms_docref_add_default_authz_policies>
 builder.Services.AddAuthorization();
+// </ms_docref_add_default_authz_policies>
 
 var app = builder.Build();
 
 app.UseHttpsRedirection();
 
+// <ms_docref_enable_authn_capabilities>
 app.UseAuthentication();
+// </ms_docref_enable_authn_capabilities>
+// <ms_docref_enable_authz_capabilities>
 app.UseAuthorization();
+// </ms_docref_enable_authz_capabilities>
 
 var summaries = new[]
 {
     "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
 };
 
+// <ms_docref_protect_endpoint>
 app.MapGet("/weatherforecast", [Authorize] () =>

```

Comment:
@ckittel @ferantivero Critical to the API tutorial and thus its code sample is teaching the developer how to use add permissions, expose scopes, and then require a scopes be present in the access token to access a method or certain data the method might return.

So, not merely an `[authorize]` attribute on the method, but also: "Does this identity (user, app) have access to this method, have they granted consent to the client app to access that data?"

@hpsin might have some good insight here.

Reply:
@mmacy: we need to use [RequiredScope("access_as_user")] here

Reply:
@jmprieur let's start people off with something better than user impersonation.  Maybe "forecast.read"?

Reply:
@hpsin : sure.
You would also need to change the app registration instructions: https://github.com/Azure-Samples/ms-identity-docs-code-csharp/tree/main/src/protect-api#register-the-web-api-application-in-your-azure-active-directory

I would also suggest the scope is set in the config file, not in the code: https://github.com/AzureAD/microsoft-identity-web/wiki/authorization-policies#with-the-requiredscopes-attribute

And also why not use the app provisioning tool to register the app? It's code first app registration, so it would find the value of the scope in the code?

Reply:
From @hpsin (paraphrased):

### API tutorial requirements

A day 1 tutorial should clearly indicate how to verify both pieces of authorization:
 
- Does the **user** have access to the API?
- Does the **app** have access to the API?
 
```csharp
// Users don't get access to this API just because they exist. Always validate that the user should have access to your API.
Validate(user.roles.includes(roles.Member));
 
// Also, applications need permission from the user to access this API.
Validate(token.scopes.contains("goodScope.name"));
```

Reply:
@hpsin : let's take this offline and provide the right API in Id.Web if it's not there already.

Alternatively, @mmacy we could use the authorization policies:
https://github.com/AzureAD/microsoft-identity-web/wiki/authorization-policies#with-authorization-policies ?

Reply:
some of the initial feedback is now applied addressed (require scopes, policies and configuration) from [3b28d](https://github.com/Azure-Samples/ms-identity-docs-code-csharp/pull/3/commits/3b28d2b122c1f4d9b20f04c9ec348076917082af). 

We plan to iterate on top of this to apply https://github.com/Azure-Samples/ms-identity-docs-code-csharp/pull/3#discussion_r747733902 and https://github.com/Azure-Samples/ms-identity-docs-code-csharp/pull/3#discussion_r747714219 as soon as we confirm these are settled.

Reply:
I'll check with @jmprieur and @hpsin and get back to you. I have some ideas on a basic API design to demonstrate authZ concepts.

Reply:
thanks @mmacy, we can decently make changes in a future PR. 

In the meantime we found no harm in going ahead with the @hpsin [feedback](https://github.com/Azure-Samples/ms-identity-docs-code-csharp/pull/3#discussion_r747714219) so it is now addressed from [ecef9](https://github.com/Azure-Samples/ms-identity-docs-code-csharp/pull/3/commits/ecef932e889748a00c346b3cb272bcb4ce0498f8) and it can be easily reverted/changed if there is any desire for that.